Assets/MixedRealityToolkit.SDK/Features/UX/Interactable/Scripts/InteractableToggleCollection.cs

```diff
@@ -11,78 +12,108 @@ namespace Microsoft.MixedReality.Toolkit.UI
     /// </summary>
     public class InteractableToggleCollection : MonoBehaviour
     {
-        [Tooltip("Interactables that will be managed by this controller")]
+        [Tooltip("Array of Interactables that will be managed by this controller")]
+        /// <summary>
+        /// Array of Interactables that will be managed by this controller
+        /// </summary>
         public Interactable[] ToggleList;
 
-        [Tooltip("Currently selected index or default starting index")]
-        public int CurrentIndex;
 
-        [Tooltip("exposed selection changed event")]
-        public UnityEvent OnSelectionEvents;
+        [Tooltip("Currently selected index in the ToggleList, default is 0")]
+        [SerializeField]
+        private int currentIndex;
 
-        private void Start()
+        /// <summary>
+        /// The current index in the array of interactable toggles
+        /// </summary>
+        public int CurrentIndex
         {
-            for (int i = 0; i < ToggleList.Length; ++i)
+            get
             {
-                int itemIndex = i;
-                // add selection event handler to each button
-                ToggleList[i].OnClick.AddListener(() => OnSelection(itemIndex));
-                ToggleList[i].CanDeselect = false;
+                return currentIndex;
+            }
+            set
+            {
+                SetSelection(value, false, true);
             }
-
-            OnSelection(CurrentIndex, true);
         }
 
-        private void OnEnable()
+        [Tooltip("This event is triggered when any of the toggles in the ToggleList are selected")]
+        /// <summary>
+        /// This event is triggered when any of the toggles in the ToggleList are selected
+        /// </summary>
+        public UnityEvent OnSelectionEvents = new UnityEvent();
+
+        private void Start()
         {
-            OnSelection(CurrentIndex, true);
+            if (ToggleList != null)
+            {
+                // Add listeners to each toggle in ToggleList
+                for (int i = 0; i < ToggleList.Length; ++i)
+                {
+                    int itemIndex = i;
+                    ToggleList[i].OnClick.AddListener(() => SetSelection(itemIndex, true, false));
```

Comment:
You need to save list of actions and then remove this list. The removelistener later is a different anonymous instance